### PR TITLE
plf_list: add version 2.57, support apple-clang 14

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **plf_list/***

- add version 2.57
- plf_list >= 2.50 doesn't support apple-clang 14 because apple-clang 14 defaults C++20
- use `add_subdirectory` in test_v1_package

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/plf_list/all/conanfile.py [hunk 1]
```diff
@@ -27,7 +28,13 @@ def source(self):
...
 
     def build(self):
-        pass
+        # apple-clang 14 defaults C++ 20, but it doesn't support C++20 features little
+        # https://www.reddit.com/r/cpp/comments/v6ydea/xcode_now_defaults_to_c20/
+        if Version(self.version) >= "2.50":
+            replace_in_file(self, os.path.join(self.source_folder, "plf_list.h"),
+            "if __cplusplus > 201704L && ((defined(_LIBCPP_VERSION) && _LIBCPP_VERSION >= 13) || !defined(_LIBCPP_VERSION)) && ((defined(__clang__) && (__clang_major__ >= 13)) || (defined(__GNUC__) && __GNUC__ >= 10) || (!defined(__clang__) && !defined(__GNUC__)))",
+            "if __cplusplus > 201704L && ((defined(_LIBCPP_VERSION) && _LIBCPP_VERSION >= 13) || !defined(_LIBCPP_VERSION)) && (!defined(__APPLE_CC__) && (defined(__clang__) && (__clang_major__ >= 13)) || (defined(__GNUC__) && __GNUC__ >= 10) || (!defined(__clang__) && !defined(__GNUC__)))"
+            )
```
Comment (SpaceIm) [inline recipes/plf_list/all/conanfile.py:37]: no_copy_source must be removed if there is a patch

Comment (toge) [reply recipes/plf_list/all/conanfile.py:37]: @SpaceIm 
I missed it.
Thank you.

Comment (SpaceIm) [reply recipes/plf_list/all/conanfile.py:36]: Is it possible to move this to a patch file please?

Comment (toge) [reply recipes/plf_list/all/conanfile.py:36]: @SpaceIm 
I agree with you.
Let me try.

Comment (toge) [reply recipes/plf_list/all/conanfile.py]: Fixed

**Review by jwillikers — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Review by SpaceIm — APPROVED** [review]

**Pull request merged**